core/src/main/java/org/apache/carbondata/core/metadata/schema/SchemaReader.java

```diff
@@ -81,10 +81,13 @@ public static TableInfo getTableInfo(AbsoluteTableIdentifier identifier)
   }
 
   public static TableInfo inferSchema(AbsoluteTableIdentifier identifier,
-      boolean isCarbonFileProvider) throws IOException {
+      boolean isCarbonFileProvider, TableInfo tableInfoFromCache) throws IOException {
     // This routine is going to infer schema from the carbondata file footer
     // Convert the ColumnSchema -> TableSchema -> TableInfo.
     // Return the TableInfo.
+    if (tableInfoFromCache != null) {
```

Comment:
If tableInfoFromCache is found, No need to call inferschema itself, please handle this outside infer schema

Reply:
Done